python/cuml/test/test_random_forest.py

```diff
@@ -808,7 +812,36 @@ def test_rf_get_json(estimator_type, max_depth, n_estimators):
     assert isinstance(json_obj, list)
     assert len(json_obj) == n_estimators
 
-    # Test 3: Traverse JSON trees and get the same predictions as cuML RF
+    # Test 3: The instance count of each node must be equal to the sum of
+    # the instance counts of its children. Note that the instance count
+    # is only available with the new backend.
+    if use_experimental_backend:
+        def check_instance_count_for_non_leaf(tree):
```

Comment:
I'm not so sure. If we were to move this function outside the test case, it may be harder to understand what's being tested in this test case. In general, we often tolerate some duplication in unit tests so that the unit tests are self-contained. See [this link](https://stackoverflow.com/a/11837973).

Reply:
I would say that as a reader I found this test hard to understand. It's over 100 lines and consists of 4 subtests snd quite a few sub-functions along with different branching for old and new. Tests 1+2 are very related to the json serialization while 3+4 are covering the new instance count functionality, which is exposed via json but is actually a bit different. Definitely found it confusing, and I think others who have to debug a future issue here would feel similarly.

Reply:
Got it. Let me create a separate unit test for the instance count functionality.

Reply:
@JohnZed Can you look at it again and see if the new test looks reasonable to you?

Reply:
I think it looks much clearer, thanks! I still look forward to getting rid of the two backend branches but to do that we need to clobber the old backend first!